### PR TITLE
Integration Specs For Views

### PR DESCRIPTION
**Pull request by luftedar**
### Hello :wave: 
# Feature: Integration Specs for Views

In this update:

- Used `Capybara` to write integration tests for each view in our project.
- Tests implemented for the following specs:
  -  Login page
  - User index page.
  - User show page.
  - Post index page.
  - Post show page.

Have a nice CR :+1:

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-    comment_creator = User.new(name: 'Jerry', photo: 'Cat.png', bio: 'bio', posts_counter: 0)
-    post.comments.create!(text: 'This is my first post', user: comment_creator)
-    post.comments.create!(text: 'This is my second post', user: comment_creator)
-    expect(post.comments.length).to eql(2)
+    it 'Check Comments Counter' do
+      current = post.comments_counter
+      comment_new = Comment.new(user:, post:, text: 'New Test Comment')
+      comment_new.update_comments_counter
+      expect(post.comments_counter).to eq(current + 1)
+    end
```
Comment (Emmyn5600) [inline spec/models/comment_spec.rb:15]: - [ ] None of your tests are passing successfully, this is due to the line 6 and 12 

(`post = Post.new(user:, title: 'test post', text: 'Testing post')`) 

 (`comment_new = Comment.new(user:, post:, text: 'New Test Comment')`)

image

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @luftedar and @gracekabaghe 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [x] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (luftedar) [reply spec/models/comment_spec.rb:15]: Hi @Emmyn5600, thanks for your reviews. The problem is I don't have any error on my side :smile:. I fixed the lines that you pointed. Hope it works. Have a nice day :100:

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 
 RSpec.describe Comment, type: :model do
-  it 'posts comments' do
-    author = User.new(name: 'Tom', photo: 'Cat.png', bio: 'bio', posts_counter: 0)
-    post = Post.new(title: 'New post', text: 'Good evening', user: author, likes_counter: 0, comments_counter: 0)
-    post.save!
+  describe 'Comment model' do
+    user1 = User.create(name: 'Orcun', photo: 'Cat.png', bio: 'Test Bio')
+    post = Post.new(user: user1, title: 'test post', text: 'Testing post')
+    comment = Comment.new(user: user1, post:, text: 'Test Comment')
```
Comment (SajjadAhmad14) [inline spec/models/comment_spec.rb:7]: Tests are failing due to an unexpected comma here without initializing the post. Kindly fix it.
fails

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
+    comment.update_comments_counter
 
-    comment_creator = User.new(name: 'Jerry', photo: 'Cat.png', bio: 'bio', posts_counter: 0)
-    post.comments.create!(text: 'This is my first post', user: comment_creator)
-    post.comments.create!(text: 'This is my second post', user: comment_creator)
-    expect(post.comments.length).to eql(2)
+    it 'Check Comments Counter' do
+      current = post.comments_counter
+      comment_new = Comment.new(user: user1, post:, text: 'New Test Comment')
```
Comment (SajjadAhmad14) [inline spec/models/comment_spec.rb:12]: Tests are failing due to an unexpected comma here without initializing the post. Kindly fix it.

fails

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Orcun,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by luftedar** [comment]
# Hello

I deleted old specs due to the unknown bug. On my side, I don't have a problem but deleted it anyway. 

Please note that new ones have already passed. Please check the following image:

hata

**Review by t-yanick — APPROVED** [review]
Hi @luftedar, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**